mentoring/tests/integration/test_dashboard.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2015 Harvard, edX & OpenCraft
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+from mock import Mock, patch
+from xblockutils.base_test import SeleniumXBlockTest
+
+
+class MockSubmissionsAPI(object):
+    """
+    Mock the submissions API, since it's not available in the test environment.
+    """
+    def __init__(self):
+        self.submissions = {}
+
+    def dict_to_key(self, dict_key):
+        return (dict_key['student_id'], dict_key['course_id'], dict_key['item_id'], dict_key['item_type'])
+
+    def create_submission(self, dict_key, submission):
+        record = dict(
+            student_item=dict_key,
+            attempt_number=Mock(),
+            submitted_at=Mock(),
+            created_at=Mock(),
+            answer=submission,
+        )
+        self.submissions[self.dict_to_key(dict_key)] = record
+        return record
+
+    def get_submissions(self, key, limit=1):
+        assert limit == 1
```

Comment:
The *mock* submissions API here only works with `limit=1`. It doesn't store enough data to be able to simulate the mock API and serve requests for anything other than the most recent item.